### ISSUE TITLE
"operator" is a registered word. Please don't use it as parameter name.

### ISSUE THREAD

**Issue by toge** [jmcnamara/libxlsxwriter#353]
In 1.1.2, `worksheet_filter_column2` has a parameter named `operator`.
Since `operator` is a C++ registered word, g++ will report this as a syntax error.
Can you please fix it?

```
include/xlsxwriter/worksheet.h:3725:67: Error: expected ‘)’ before ‘operator’
 3725 |                                    lxw_filter_rule *rule2, uint8_t operator);
      |                                                                   ^~~~~~~~~
      |                                                                   )
```

**Comment by jmcnamara** [comment]
Thanks.

I've pushed a fix to main, if you can test it.

P.S., how did you compile with c++/g++ to find this issue?

**Comment by jmcnamara** [comment]
Fixed in version 1.1.3.

**Comment by toge** [comment]
Thanks for the quick fix.
I have confirmed that 1.1.3 can be compiled.

I found this problem with gcc 11.2.1 on Fedora34.
